### PR TITLE
Parse pragma solidity version numbers

### PR DESCRIPTION
**Pull request by seanyoung**
Many Solidity features depend on the version of the compiler. So, parse the version pragma and in another PR we will use this information.

This PR adds pragmas to the ast and we test it with a graphviz dot file.

### CODE REVIEW

File: solang-parser/src/helpers/fmt.rs [hunk 1]
```diff
@@ -803,14 +803,79 @@ impl Display for pt::SourceUnitPart {
...
+            Self::PragmaDirective(inner) => inner.fmt(f),
+            Self::StraySemicolon(_) => f.write_char(';'),
+        }
+    }
+}
+
+impl Display for pt::PragmaDirective {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        match self {
+            Self::Identifier(_, ident, val) => {
```
Comment (seanyoung) [inline solang-parser/src/helpers/fmt.rs:815]: I don't understand why codecov thinks this is not covered - there is literally a test for this in the same file.

Maybe the parser tests are included in coverage?

**Comment by seanyoung** [comment]
@LucasSte @xermicus let me know if you're interested in reviewing this, else I will merge it

**Comment by LucasSte** [comment]
> @LucasSte @xermicus let me know if you're interested in reviewing this, else I will merge it

I'm reviewing it.

File: tests/contract.rs [hunk 2]
```diff
@@ -183,4 +187,32 @@ fn check_diagnostics(path: &Path, ns: &Namespace) {
     assert!(!found.is_empty());
 
     assert_eq!(found, expected, "source: {}", path.display());
+
+    Ok(())
+}
+
+fn check_dot(path: &Path, ns: &Namespace) -> io::Result<()> {
```
Comment (LucasSte) [inline tests/contract.rs:194]: Why are you adding graphviz again?

Comment (seanyoung) [reply tests/contract.rs:194]: This means that the graphviz code paths are tested, and the pragma ast parts are tested.

**Pull request merged**